examples/image_classifier/resnet_18/debugging_backend/test_handler.py

```diff
@@ -0,0 +1,66 @@
+import argparse
+import io
+import logging
+from pathlib import Path
+
+import torch
+
+from ts.torch_handler.image_classifier import ImageClassifier
+from ts.torch_handler.unit_tests.test_utils.mock_context import MockContext
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.DEBUG)
+
+CURR_FILE_PATH = Path(__file__).parent.absolute()
+REPO_ROOT_DIR = CURR_FILE_PATH.parent.parent.parent.parent
+EXAMPLE_ROOT_DIR = REPO_ROOT_DIR.joinpath("examples", "image_classifier", "resnet_18")
+TEST_DATA = REPO_ROOT_DIR.joinpath("examples", "image_classifier", "kitten.jpg")
+MODEL_PT_FILE = "resnet-18.pt"
+
+
+def prepare_data(batch_size):
+    """
+    Function to prepare data based on the desired batch size
+    """
+    f = io.open(TEST_DATA, "rb", buffering=0)
+    read_data = f.read()
+    data = []
+    for i in range(batch_size):
+        tmp = {}
+        tmp["data"] = read_data
+        data.append(tmp)
+    return data
+
+
+def main(batch_size):
```

Comment:
Can you convert this into a unit test runnable with pytest (we can still keep the main if people come along and run it separately) and encourage people in the doc to write a unit test for their handler?

Reply:
How about if I make an additional test in the  pytest folder and link as that an example to encourage people to write tests. I want to keep this separate for debugging. As someone who is not the most comfortable with all pytest syntax, having this example in pytest fodler, it would probably make me less likely to use it to debug.

Actually, I guess I can create a separate PR for the pytest example. The goal of that PR would be encourage everyone to write a test when they push an example. This test would also need creation of mar file, validation of output etc. What do you think?

Reply:
@agunapal Actually, I did not mean to move it into the unit test folder. What I mean is that we rename main into test_resnet_something so that we can also run it with pytest __as well__. Then you can show users: "Here try to run this with pytest test_handler.py and there goes your unit test." I would argue that unit tests are a great way to debug your handler. Think about an assert in pytest. When you assert in python you only get: "This test failed". In pytest you get: "This test failed because the objects differ in this specific way" which is very valuable I think. And because we have a single test in a not unit test folder the syntax is minimal.

Reply:
@mreso I get your point and I will make a separate PR for pytest. I tried running this as a pytest, it expects me to a fixture for batch_size.  
I am specifically trying to address something like this https://github.com/pytorch/serve/issues/2223

I wanted this example to be a low barrier for entry kind of example. I also included a screenshot of the debugger. I am hoping TorchServe customers use this with a debugger

Reply:
@agunapal just give batch_size a default value:
```
def test_main(batch_size=1):
```
Then is works in debugger and pytest without any scary syntax.

Reply:
Thanks! That works. Added pytest info in the README as well